sound/soc/sof/sof-audio.c

```diff
@@ -92,11 +129,20 @@ static int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swid
 		.id = swidget->comp_id,
 	};
 	struct sof_ipc_reply reply;
-	int ret;
+	int ret = 0;
+
+	/* decrement usecount and update config for static DAI widgets */
+	if (swidget->use_count > 1) {
+		if (!swidget->dynamic_pipeline_widget)
+			ret = sof_static_widget_config(sdev, swidget);
+
+		swidget->use_count--;
+		return ret;
+	}
 
 	/* only free when refcount is 0 */
 	if (--swidget->use_count)
```

Comment:
mmh, I'll blame this on GH. This comment was for the previous version of the last commit for this hunk:
```
@@ -92,11 +129,20 @@ static int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swid
 		.id = swidget->comp_id,
 	};
 	struct sof_ipc_reply reply;
-	int ret;
+	int ret = 0;
+
+	/* decrement usecount and update config for static DAI widgets */
+	if (swidget->use_count > 1) {
+		if (!swidget->dynamic_pipeline_widget)
+			ret = sof_static_widget_config(sdev, swidget);
+
+		swidget->use_count--;
+		return ret;
+	}
 
 	/* only free when refcount is 0 */
 	if (--swidget->use_count)
-		return;
+		return 0;
 
 	if (!swidget->private)
 		return 0;
```
There you only get past the first `if (swidget->use_count > 1)` if the `.use_count` equals 1 or 0. Then the next condition `if (--swidget->use_count)` isn't satisfied if `.use_count` was 1 and hopefully it wouldn't even be tried if it was 0, because you probably don't want it negative. But you modified that code now, so the comment isn't relevant any more.